src/theory/strings/term_registry.h

```diff
@@ -232,8 +232,19 @@ class TermRegistry : protected EnvObj
    */
   Node mkNConcat(const std::vector<Node>& c, TypeNode tn) const;
 
-  /** compute relevant terms of the theory of strings */
-  void getRelevantTermSet(std::set<Node>& termSet);
+  /**
+   * Compute relevant terms of the theory of strings. This is called once
+   * per full effort check by the base solver at the beginning of checkInit.
+   */
+  void computeRelevantTermSet();
+  /**
+   * Get the relevant term set, returns the set computed by the above function.
+   *
+   * The relevant term set is all terms that belong to theory of strings that
+   * appear in the current set of assertions, or are marked as shared terms
+   * for the theory of strings.
+   */
+  const std::set<Node>& getRelevantTermSet() const;
```

Comment:
Is this separation between `compute` and `get` standard in our codebase? I am a bit worried of someone calling `get` while forgetting to call `compute` before. Can we protect ourselves against something like this using some assertion inside `get`?

Reply:
Yes, I tend to use this naming scheme.

The comment mentions when this is computed, with the assumption that you can only call `getRelevantTermSet` during a full effort check.  I can add some further assertions to ensure this method is being used properly.

Reply:
I've refactored and improved this, PTAL.